docs/commands.md

```diff
@@ -1,150 +1,538 @@
+```
+RG.CONFIGSET <key> <value> [<key> <value> ...]
+```
 
+_Arguments_
+
+* _key_: the configuration option to set
+* _value_: the new value
+
+_Return_
+
+An array with an entry per key. The entry is simple 'OK' string, or an error if the option can't be configured in runtime.
+
+**Examples**
 
-## RG.GETRESULTSBLOCKING
-```sql
-RG.GETRESULTSBLOCKING execution-id
 ```
-* execution-id: the execution id for which to return the results.
+127.0.0.1:6379> RG.CONFIGSET ProfileExecutions 1
+1) OK
+127.0.0.1:6379> RG.CONFIGSET foo bar
+1) OK - value was saved in extra config dictionary
+127.0.0.1:6379> RG.CONFIGGET foo
+1) "bar"
+```
 
-Same as `RG.GETRESULTS` but will block untill the execution is finished in which case it will return the Execution Result.
+## RG.DROPEXECUTION
+The **RG.DROPEXECUTION** command removes the [execution](functions.md#execution) of a function from the executions list.
+
+**Redis API**
+
+```
+RG.DROPEXECUTION <id>
+```
 
-### Returns
-The [Execution Result](#execution-result) or an error when:
+_Arguments_
 
-* the execution-id doesn't exist
-* execution is no longer available (see configuration, we keep by default the last 1000 results)
+* _id_: the [execution ID](functions.md#executionid) to remove
 
+_Return_
+
+A simple 'OK' if successful, or an error if the execution does not exist or is still running.
+
+**Examples**
+
+```
+127.0.0.1:6379> RG.DROPEXECUTION 0000000000000000000000000000000000000000-1
+OK
+```
 
 ## RG.DUMPEXECUTIONS
-```sql
+The **RG.DUMPEXECUTIONS** command outputs the list of [function executions](functions.md#execution). The executions list's length is capped by the [MaxExecutions](configuration.md#maxexecutions) configuration option.
+
+**Redis API**
+
+```
 RG.DUMPEXECUTIONS
 ```
 
-Return all the executions.
+_Return_
 
-### Returns
-List of executions with
-* executionId
-* status (running/done)
+An array with an entry per execution. Each entry is made of alternating key name and value entries as follows:
+
+* **executionId**: the [execution ID](functions.md#executionid)
+* **status**: the status (e.g. 'pending', 'running', 'done'?)
+
+**Examples**
+
+```
+127.0.0.1:6379> RG.DUMPEXECUTIONS
+1) 1) "executionId"
+   2) "0000000000000000000000000000000000000000-0"
+   3) "status"
+   4) "done"
+2) 1) "executionId"
+   2) "0000000000000000000000000000000000000000-1"
+   3) "status"
+   4) "running"
+```
+
+## RG.DUMPREGISTRATIONS
+The **RG.DUMPREGISTRATIONS** command outputs the list of [function registrations](functions.md#registration).
+
+**Redis API**
+
+```
+RG.DUMPREGISTRATIONS
+```
+
+_Return_
+
+An array with an entry per registration. Each entry is made of alternating key name and value entries as follows:
+
+* **id**: the [registration ID](functions.md#registrationid)
+* **reader**: the [reader](readers.md)
+* **desc**: the description
+* **RegistrationData**: an array of the following:
```

Comment:
Actually the RegistrationData might be different between reader, today its the same but its the reader responsibility to reply on this part. I plan to create a one base reader that all the reader will user and there will be a common part and a different part under this section. @itamarhaber do you think we should leave it as is for now?

Reply:
For now, yes, as we're documenting the current behavior. As for revamping the API, I totally agree. Generally, I'd like to see something like this for run and the rest.

```python
class GearsBuilder.register(**kwargs)
```